### PR TITLE
Add support for PT8211 DAC

### PR DESCRIPTION
**Pull request by whc2001**
**UNTESTED!!!** It compiles, but it's not tested with real hardware yet as I'm physically away from my home. There is a small possibility that it just does not work at all.

- Added a flag (`CONFIG_I2S_USE_MSB_FORMAT`) for using MSB format instead of standard Phillips format for I2S. This flag will be triggered by the selection of the PT8211 audio board.
    - Note that this is not exactly the same as the LSBJ format PT8211 demanded. It's only that their timing is equivalent under the native bit-depth (PT8211 is 16 bit, and this project also uses 16 bit). The [Arduino I2S library](https://github.com/schreibfaul1/ESP32-audioI2S/blob/1d9b299d0197f99fc70335295adcd226dc220f88/src/Audio.cpp#L4929) is doing this so I assume it's acceptable.
- Added PT8211 audio board definition. I have included a mute output pin with configurable GPIO number and enable level, despite PT8211 don't even have any kind of control input. It's intended for something like an external amplifier.

### CODE REVIEW

File: components/audio_board/Kconfig.projbuild [hunk 1]
```diff
@@ -274,6 +287,24 @@ menu "Audio Board"
 	                This is labeled "SD" on chip/boards.
 	    endmenu
 
+		menu "PT8211 interface Configuration"
+	        depends on DAC_PT8211
+
+	        config PT8211_MUTE_PIN
+	            int "Master mute/unmute"
+	            default 18
```
Comment (CarlosDerSeher) [inline components/audio_board/Kconfig.projbuild:295]: Is there a reason why you chose this as a default? Shouldn't it be -1?

Comment (whc2001) [reply components/audio_board/Kconfig.projbuild:295]: Ah good point. I just copied most of the code from another existing audio board. Also I think I should check for -1 before initializing and setting the GPIO...

**Comment by CarlosDerSeher** [comment]
Looks good but I'd prefer go try with real hardware before we merge

**Comment by whc2001** [comment]
> Looks good but I'd prefer to try with real hardware before we merge

Great, thanks! I'm gonna do that ASAP after back home.

**Comment by whc2001** [comment]
Tested. There is a slight hiss but it goes away while RESET is pressed, maybe it's WiFi interference.

https://github.com/CarlosDerSeher/snapclient/assets/16266909/760005c6-78a7-4f50-8082-76eb1f8dae64

**Comment by CarlosDerSeher** [comment]
@whc2001 Could be interference as your setup isn't perfect with the antenna that close to the analog part. What you could try is lowering wifi output power from menuconfig and see if that improves quality.

**Comment by whc2001** [comment]
> @whc2001 Could be interference as your setup isn't perfect with the antenna that close to the analog part. What you could try is lowering wifi output power from menuconfig and see if that improves quality.

Yes. I have tried to lower the TX power from 20dBm to 10dBm and the noise mostly went away.

**Comment by CarlosDerSeher** [comment]
I guess we are done here then. Thanks for your contribution!

**Pull request merged**